yt_dlp/networking/impersonate.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from abc import ABC
+from dataclasses import dataclass, fields
+from typing import Any
+
+from .common import RequestHandler, register_preference
+from .exceptions import UnsupportedRequest
+from ..compat.types import NoneType
+from ..utils import classproperty
+from ..utils.networking import std_headers
+
+
+@dataclass(order=True, frozen=True)
+class ImpersonateTarget:
+    """
+    A target for browser impersonation.
+
+    Parameters:
+    @param client: the client to impersonate
+    @param version: the client version to impersonate
+    @param os: the client OS to impersonate
+    @param os_vers: the client OS version to impersonate
+
+    Note: None is used to indicate to match any.
+    """
+    client: str | None = None
+    version: str | None = None
+    os: str | None = None
+    os_vers: str | None = None
```

Comment:
```suggestion
    os_ver: str | None = None
```
`os_vers` is typically read "os version**s**"